src/ccs/cccs.rs

```diff
@@ -28,47 +28,55 @@ use std::sync::Arc;
 
 use super::util::compute_sum_Mz;
 use super::util::virtual_poly::VirtualPolynomial;
-use super::{CCSShape, CCSWitness};
+use super::CCS;
 
 /// A type that holds the shape of a Committed CCS (CCCS) instance
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(bound = "")]
-pub struct CCCSShape<G: Group> {
-  // Sequence of sparse MLE polynomials in s+s' variables M_MLE1, ..., M_MLEt
-  pub(crate) M_MLE: Vec<MultilinearPolynomial<G::Scalar>>,
-
-  pub(crate) ccs: CCSShape<G>,
+pub struct CCCS<G: Group> {
+  // The `z` vector used as input for this instance.
+  pub(crate) z: Vec<G::Scalar>,
+  // Commitment to the witness of `z`.
+  pub(crate) w_comm: Commitment<G>,
 }
 
-/// A type that holds a CCCS instance
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(bound = "")]
-pub struct CCCSInstance<G: Group> {
-  // Commitment to a multilinear polynomial in s' - 1 variables
-  pub(crate) C: Commitment<G>,
+impl<G: Group> CCCS<G> {
+  /// Generates a new CCCS given a reference to it's original CCS repr and it's public and private inputs.
+  pub(crate) fn new(
+    ccs: &CCS<G>,
+    ccs_matrix_mle: &[MultilinearPolynomial<G::Scalar>],
```

Comment:
> Or the css_matrix_mle is carried from other upper level logic and here is just reusing that already computed data?

That's exactly what happens. Ideally, I'll have the mles (as well as the CCS) serialized or given by the wire. So I don't want to spend time computing MLES. Specially since the matrixes are sparse. Which has the idea to minimize memory usage.